From 9247b48a96771a30371c635a5386a74bcb219257 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Thu, 2 Feb 2023 22:24:14 +0200
Subject: [PATCH 3/9] unpopulated-alloc.c: Drop restriction for
 DMA_BIT_MASK(64)

The single at the monent in-tree user of that API sets DMA_BIT_MASK(64),
so this is why this restriction is here.

But there is an out-of-tree user (PVRKM) which sets DMA_BIT_MASK(40),
so drop the restriction for now.
In general it is not good to make an excuse for out-of-tree modules...

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 drivers/xen/unpopulated-alloc.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/xen/unpopulated-alloc.c b/drivers/xen/unpopulated-alloc.c
index 3988480d97be..66c19aeca54c 100644
--- a/drivers/xen/unpopulated-alloc.c
+++ b/drivers/xen/unpopulated-alloc.c
@@ -287,10 +287,6 @@ EXPORT_SYMBOL(xen_free_unpopulated_pages);
 int xen_alloc_unpopulated_contiguous_pages(struct device *dev,
 		unsigned int nr_pages, struct page **pages)
 {
-	/* XXX Handle devices which support 64-bit DMA address only for now */
-	if (dma_get_mask(dev) != DMA_BIT_MASK(64))
-		return -EINVAL;
-
 	return alloc_unpopulated_pages(nr_pages, pages, true);
 }
 EXPORT_SYMBOL(xen_alloc_unpopulated_contiguous_pages);
-- 
2.34.1

